model_api/cpp/models/src/image_model.cpp

```diff
@@ -55,137 +55,31 @@ RESIZE_MODE ImageModel::selectResizeMode(const std::string& resize_type) {
-        if (!model->has_rt_info<std::string>("model_info", "labels")) {
-            throw std::runtime_error("Configuration or model rt_info should contain labels"); //TODO
-        }
-        labels = split(model->get_rt_info<std::string>("model_info", "labels"), ' ');
-    } else {
-        labels = labels_iter->second.as<std::vector<std::string>>();
-    }
-
-    if (model->has_rt_info("model_info", "embedded_processing")) {
-        embedded_processing = model->get_rt_info<bool>("model_info", "embedded_processing");
-    }
-
-    if (model->has_rt_info("model_info", "orig_width")) {
-        netInputWidth = model->get_rt_info<size_t>("model_info", "orig_width");
-    }
-    if (model->has_rt_info("model_info", "orig_height")) {
-        netInputHeight = model->get_rt_info<size_t>("model_info", "orig_height");
-    }
-
-    auto pad_value_iter = configuration.find("pad_value");
-    if (pad_value_iter == configuration.end()) {
-        if (model->has_rt_info("model_info", "pad_value")) {
-            // get_rt_info() incorrectly casts to uint8_t. Cast through int
-            int decoded = model->get_rt_info<int>("model_info", "pad_value");
-            if (0 > decoded || 255 < decoded) {
-                throw std::runtime_error("pad_value must be in range [0, 255]");
-            }
-            pad_value = decoded;
-        }
-    } else {
-        pad_value = pad_value_iter->second.as<uint8_t>();
-    }
-
-    reverse_input_channels = get_from_any_maps("reverse_input_channels", configuration,
-                                               model->has_rt_info("model_info") ? model->get_rt_info<ov::AnyMap>("model_info") : ov::AnyMap{},
-                                               reverse_input_channels);
-
-    auto scale_values_iter = configuration.find("scale_values");
-    if (scale_values_iter == configuration.end()) {
-        if (model->has_rt_info("model_info", "scale_values")) {
-            scale_values = model->get_rt_info<std::vector<float>>("model_info", "scale_values");
-        }
-    } else {
-        scale_values = labels_iter->second.as<std::vector<float>>();
-    }
-    auto mean_values_iter = configuration.find("mean_values");
-    if (mean_values_iter == configuration.end()) {
-        if (model->has_rt_info("model_info", "mean_values")) {
-            mean_values = model->get_rt_info<std::vector<float>>("model_info", "mean_values");
-        }
-    } else {
-        mean_values = labels_iter->second.as<std::vector<float>>();
-    }
+    labels = get_from_any_maps("labels", top_priority, mid_priority, labels);
+    embedded_processing = get_from_any_maps("embedded_processing", top_priority, mid_priority, embedded_processing);
+    netInputWidth = get_from_any_maps("orig_width", top_priority, mid_priority, netInputWidth);
+    netInputHeight = get_from_any_maps("orig_height", top_priority, mid_priority, netInputHeight);
+    pad_value = get_from_any_maps("pad_value", top_priority, mid_priority, pad_value);
```

Comment:
The removed code contained some typecasting and validation for `pad_value`, is that now handled in `get_from_any_maps`? Or is it no longer necessary?